ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/PostCommitValidation.scala

```diff
@@ -90,12 +93,13 @@ private[appendonlydao] object PostCommitValidation {
       val referredContracts = collectReferredContracts(transaction, divulged)
       if (referredContracts.isEmpty) {
         None
-      } else {
-        contractStorageBackend
-          .maximumLedgerTime(referredContracts)(connection)
-          .map(validateCausalMonotonicity(_, transactionLedgerEffectiveTime))
-          .getOrElse(Some(Rejection.UnknownContract))
-      }
+      } else
+        contractStorageBackend.maximumLedgerTime(referredContracts)(connection) match {
+          case Failure(MissingContracts(missingContractIds)) =>
+            Some(Rejection.UnknownContracts(missingContractIds.map(_.coid)))
+          case Failure(_) => Some(Rejection.UnknownContract)
```

Comment:
It seems like we use `UnknownContract` as a replacement for any unknown "internal error"-like exceptions. Do we have a better option for this case?

Reply:
Yes, this is just preserving existing functionality without converting the response of maximumLedgerTime in the `Either[Set[ContractId], ...]

Reply:
Ok, it is misleading like this. I'll address

Reply:
Done 87c9b918072712fed1b6224060a2b39f34cf6feb